docs/src/main/sphinx/release/release-354.md

```diff
@@ -0,0 +1,56 @@
+# Release 354 (XXXXXXXX Mar 2021)
+
+## General
+
+* Preferred write partitioning is now triggered automatically when the number
+  of written partitions exceeds or is equal to
+  ``preferred-write-partitioning-min-number-of-partitions``. The number of written
+  partitions is estimated from stats, therefore CTAS or insert source tables needs to have
+  statistics collected. To forcefully enable preferred partitioning,
+  ``preferred-write-partitioning-min-number-of-partitions`` needs to be set to
+  ``0`` or ``1``. To fully disable preferred write partitioning, set
+  ``use-preferred-write-partitioning`` to ``false``.
+  ({issue}`6920`)
+* Improve performance of LEFT, RIGHT and FULL JOIN when one side of the join is known to produce a single row. ({issue}`7090`)
+  
+## Security
+
+* Fix `Oauth2` repeated authentication, after previous token expires. ({issue}`7172`)
```

Comment:
Is this a bugfix or perf improvement? might be good to elaborate on user visible impact if there is. cc @kokosing

Reply:
It is a bugfix. Previously when token expired we were not able to do full authentication flow successfully.

@s2lomon ?

Reply:
How about 
```
* Fix retries for OAuth 2.0 authentication in case of token expiry. ({issue}`7172`)
```

(If others think the current wording is fine, I'm okay with merging as is.)